examples/shred.py

```diff
@@ -0,0 +1,63 @@
+# Examples: https://github.com/mate-desktop/python-caja/tree/master/examples
+#
+# Note: Currently there is no Python-Caja example available to configure the
+# number of shred iterations via the Preferences, Configure Extension dialog.
+# https://github.com/mate-desktop/python-caja/issues/71
+
+import os
+
+import gi
+from gi.repository import Caja, GObject, Gio, Gtk
+
+
+class ShredMenuProvider(GObject.GObject, Caja.MenuProvider):
+    # Hard-coded settings
+    SHRED_ITERATIONS = 0
+    SHRED_EXECUTABLE = 'shred'
+    SHRED_ICON = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'icons/shred.png')
+
+    def __init__(self):
+        self.shred_args = '-uz -n {}'.format(self.SHRED_ITERATIONS)
+
+    def menu_activate_cb(self, menu, files):
+        if not files or not len(files):
+            return
+
+        dialog = Gtk.MessageDialog(
+            parent=None,
+            flags=0,
+            message_type=Gtk.MessageType.QUESTION,
+            buttons=Gtk.ButtonsType.YES_NO,
+            text="Are you sure to shred files?",
+        )
+        dialog.format_secondary_text("WARNING: This cannot be undone!")
+        response = dialog.run()
+        dialog.destroy()
+        if response == Gtk.ResponseType.NO:
+            return
+
+        shred_files = ''
+        for shred_file in files:
+            shred_files += '"{}" '.format(shred_file.get_location().get_path())
+  
+        # Start Shred
+        cmd = '{} {} {}'.format(self.SHRED_EXECUTABLE, self.shred_args, shred_files)
+
+        # Save command for debugging
+        # with open('/tmp/shred.txt', 'w') as f:
+        #     f.write(cmd)
+
+        # Start Shred command
+        os.system(cmd)
```

Comment:
This is dangerous.

Since you're doing a non-reversible destructive operation, make sure to wrap the action inside a "Yes" clause. Otherwise pressing the Escape key in the dialog will still delete the file.

```suggestion
        if response == Gtk.ResponseType.YES:
            shred_files = ''
            for shred_file in files:
                shred_files += '"{}" '.format(shred_file.get_location().get_path())

            # Start Shred
            cmd = '{} {} {}'.format(self.SHRED_EXECUTABLE, self.shred_args, shred_files)

            # Start Shred command
            os.system(cmd)
```

Reply:
Fixed in https://github.com/mate-desktop/python-caja/pull/72/commits/78bc54d1ae510f7aa21c1ae55864ebba7c038d60